File: fullstackwithflutter_ui/lib/models/activity.dart
class Activity {
  final int id;
  final String? type;
  final String? description;
  final int? userId;
  final String? userName;
  final DateTime createdDate;
  final String? details;
  final String? icon;
  final String? color;

  Activity({
    required this.id,
    this.type,
    this.description,
    this.userId,
    this.userName,
    required this.createdDate,
    this.details,
    this.icon,
    this.color,
  });

  factory Activity.fromJson(Map<String, dynamic> json) {
    return Activity(
      id: json['id'],
      type: json['type'],
      description: json['description'],
      userId: json['userId'],
      userName: json['userName'],
      createdDate: DateTime.parse(json['createdDate']),
      details: json['details'],
      icon: json['icon'],
      color: json['color'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'type': type,
      'description': description,
      'userId': userId,
      'userName': userName,
      'createdDate': createdDate.toIso8601String(),
      'details': details,
      'icon': icon,
      'color': color,
    };
  }
}

File: FullstackWithFlutter/FullstackWithFlutter.Core/Models/Activity.cs
﻿namespace FullstackWithFlutter.Core.Models
{
    public class Activity
    {
        public int Id { get; set; }
        public string? Type { get; set; } // UserRegistration, AppointmentCreation, AppointmentStatusChange, DoctorAssignment, etc.
        public string? Description { get; set; }
        public int? UserId { get; set; }
        public string? UserName { get; set; }
        public DateTime CreatedDate { get; set; }
        public string? Details { get; set; } // JSON formatında ek detaylar
        public string? Icon { get; set; } // Frontend'de gösterilecek ikon
        public string? Color { get; set; } // Frontend'de gösterilecek renk
    }
}
